### ISSUE TITLE
Support custom signature format

### ISSUE THREAD

**Issue by XiepengLi** [stanfordnlp/dspy#1244]
```
Answer questions with short factoid answers.

---

Follow the following format.

Question: ${question}
Question's Answer: often between 1 and 5 words

---

Question:
```

This is an extremely opinionated format

**Comment by theta-lin** [comment]
I agree that fixing a prompt format is not a good design choice. I recommend referring to #409 for this issue.

**Comment by okhat** [comment]
Adapters fix this in DSPY 2.5 https://github.com/stanfordnlp/dspy/blob/main/examples/migration.ipynb
